File: LeetCode/169.majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int majority = 0,cnt = 0;
        for (int i = 0;i < nums.size();i++){
            if (cnt == 0){
                majority = nums[i];
                cnt++;
            }else if (majority == nums[i]){
                cnt++;
            }else{
                cnt--;
            }
        }
        return majority;
    }
};

File: LeetCode/0169.majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int length = nums.length;
        int val = nums[0];
        int cnt = 1;
        int index = 1;
        while (index < length){
            if (val != nums[index]){
                cnt--;
                if (cnt == 0){
                    val = nums[index];
                    cnt = 1;
                }
            }else{
                cnt++;
            }
            index++;
        }
        return val;
    }
}